Assets/Scripts/UI/XboxRemap.cs

```diff
@@ -47,10 +49,36 @@ public void InitiateButton(int passed)
         button = GameObject.Find("Managers").GetComponent<InputManager>().buttons[passed].xboxButton;
         keyName = GameObject.Find("Managers").GetComponent<InputManager>().buttons[passed].xboxButton.ToString();
     }
-    //Curently not working
+    public void Remaping()
+    {
+        StartCoroutine(timerRemaping());
+    }
     public void SetButton(XboxController.XboxButton passed)
     {
+        List<string> temp = new List<string>();
+        temp = GameObject.Find("Player 1 Camera").GetComponentInChildren<PauseMenu>().xboxCodes;
+        player = GameObject.Find("PlayerDefault").GetComponentInChildren<IInputPlayer>();
+        foreach (string xKey in temp)
+        {
+            if (passed.ToString() == xKey)
+            {
+                return;
+            }
+        }
+        InputManager.RemapXboxButton(action, passed, player);
+        temp.Remove(keyName);
+        keyName = passed.ToString();
+        temp.Add(keyName);
```

Comment:
wait whats the point of modifying the temp collection, you're not doing anything with it after?

Reply:
this ends up modifying the xboxCodes list in PauseMenu so that if they remap the button the old value gets removed and the new one is added

Reply:
Oh what the hell
OOP is awful, I didn't realize that the entire collection was being passed by reference
You should definitely rename the variable from temp since it's not temporary at all
But I guess this code doesn't really matter or need to be maintainable